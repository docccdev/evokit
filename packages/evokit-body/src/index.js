import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
