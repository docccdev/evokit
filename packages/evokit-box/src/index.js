// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
