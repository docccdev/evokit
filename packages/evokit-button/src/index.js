import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
