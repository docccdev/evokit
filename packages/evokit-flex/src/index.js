import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'direction',
    'display',
    'items',
    'wrap',
    ['content', [
        'content-align',
        'content-justify',
    ]],
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'display',
    'order',
]);

Flex.Item = FlexItem;
