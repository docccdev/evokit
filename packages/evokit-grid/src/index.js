import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
