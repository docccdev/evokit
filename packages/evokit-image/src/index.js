import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
