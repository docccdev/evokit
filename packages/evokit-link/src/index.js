import { createBlock } from 'evokit';

export const Link = createBlock('a', 'link', [
    'color',
    'display',
    'lheight',
    'size',
    'style',
    'valign',
    'weight',
    'wrap',
]);
