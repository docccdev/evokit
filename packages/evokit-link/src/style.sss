@import './mixins.sss'

@custom-media --ek-link-media-small only screen and (min-width: 480px)
@custom-media --ek-link-media-medium only screen and (min-width: 768px)
@custom-media --ek-link-media-large only screen and (min-width: 960px)
@custom-media --ek-link-media-wide only screen and (min-width: 1200px)
@custom-media --ek-link-media-huge only screen and (min-width: 1400px)

:root
    // base
    --ek-link-line-height: 1em
    // prop 'link-size'
    --ek-link-size-small: 11px
    --ek-link-size-default: 13px
    --ek-link-size-big: 15px
    --ek-link-size-h1: 24px
    --ek-link-size-h2: 22px
    --ek-link-size-h3: 20px
    --ek-link-size-h4: 18px
    --ek-link-size-h5: 16px
    --ek-link-size-h6: 14px
    // prop 'link-lheight'
    --ek-link-line-height-default: var(--ek-link-line-height)
    --ek-link-line-height-medium: 1.25em
    --ek-link-line-height-large: 1.5em

.link
    display: inline
    color: inherit
    line-height: var(--ek-link-line-height)
    text-decoration: none

    @mixin hover
        cursor: pointer
        text-decoration: underline

/**
 * Display:
 *
 *  - link_display_inline
 *  - link_display_none
 *
 **/

@each $value in (inline, none)
    @mixin media .link_display_$(value)
        display: $value

/**
 * Font size:
 *
 *  - link_size_*
 *
 **/

@each $value in (small, default, big, h1, h2, h3, h4, h5, h6)
    @mixin media .link_size_$(value)
        font-size: var(--ek-link-size-$(value))

/**
 * Font weight:
 *
 *  - link_weight_*
 *
 **/

@each $value in (default, thin, light, medium, bold, black)
    @mixin media .link_weight_$(value)
        @if $value == default
            font-weight: 400
        @else if $value == thin
            font-weight: 100
        @else if $value == light
            font-weight: 300
        @else if $value == medium
            font-weight: 500
        @else if $value == bold
            font-weight: 700
        @else if $value == black
            font-weight: 900

/**
 * Valign:
 *
 *  - link_valign_*
 *
 **/

@each $value in (baseline, bottom, middle, sub, super, text-bottom, text-top, top)
    @mixin media .link_valign_$(value)
        vertical-align: $value

/**
 * Line height:
 *
 *  - link_lheight_*
 *
 **/

@each $value in (default, medium, large)
    @mixin media .link_lheight_$(value)
        line-height: var(--ek-link-line-height-$(value))

/**
 * Style:
 *
 *  - link_style_*
 *
 **/

.link_style_dotted
    border-bottom: 1px dotted

    @mixin hover
        text-decoration: none
        border-bottom-color: transparent

.link_style_multi-line
    position: relative
    display: inline-block

    &:after
        content: ''
        position: absolute
        left: 0
        top: 0
        right: 0
        bottom: 0
