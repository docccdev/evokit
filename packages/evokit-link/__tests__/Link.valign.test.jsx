import React from 'react';
import { shallow } from 'enzyme';
import { Link } from '../src';

describe('<Link link-valign="..." />', () => {
    it('props link-valign', () => {
        const wrapper = shallow(<Link link-valign='value' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<a class="ek-link ek-link_valign_value"></a>');
    });
});
