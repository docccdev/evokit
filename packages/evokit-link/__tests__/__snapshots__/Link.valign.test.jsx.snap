// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link link-valign="..." /> props link-valign 1`] = `
<a
  className="ek-link ek-link_valign_value"
/>
`;
